WooCommerce/Classes/POS/Analytics/WooAnalyticsEvent+PointOfSale.swift

```diff
@@ -77,81 +115,132 @@ extension WooAnalyticsEvent {
-                              properties: [
-                                Key.itemListType: itemListType.analyticsValue
-                              ])
+            WooAnalyticsEvent(
+                statName: .pointOfSalePreSearchRecentTermTapped,
+                properties: [Key.source: Source(itemListType: itemListType).rawValue]
+            )
         }
 
-        static func pointOfSaleItemsNextPageLoaded(itemType: POSItemType, searching: Bool) -> WooAnalyticsEvent {
-            WooAnalyticsEvent(statName: .pointOfSaleItemsNextPageLoaded,
-                              properties: [
-                                Key.itemListType: itemType.analyticsValue,
-                                Key.search: searching
-            ])
+        static func itemsPullToRefresh(
+            source: WooAnalyticsEvent.PointOfSale.Source,
+            sourceType: WooAnalyticsEvent.PointOfSale.SourceType
+        ) -> WooAnalyticsEvent {
+            WooAnalyticsEvent(
+                statName: .pointOfSaleItemsPullToRefresh,
+                properties: [Key.source: source.rawValue, Key.sourceType: sourceType.rawValue]
+            )
+        }
+
+        static func itemsNextPageLoaded(
+            source: WooAnalyticsEvent.PointOfSale.Source,
+            sourceType: WooAnalyticsEvent.PointOfSale.SourceType
+        ) -> WooAnalyticsEvent {
+            WooAnalyticsEvent(
+                statName: .pointOfSaleItemsNextPageLoaded,
+                properties: [Key.source: source.rawValue, Key.sourceType: sourceType.rawValue]
+            )
         }
 
         static func pointOfSaleSearchRemoteResultsFetched(itemType: POSItemType,
                                                           resultsCount: Int,
                                                           millisecondsSinceRequestSent: Int) -> WooAnalyticsEvent {
             WooAnalyticsEvent(statName: .pointOfSaleSearchRemoteResultsFetched,
                               properties: [
-                                Key.itemListType: itemType.analyticsValue,
+                                Key.source: Source(itemType: itemType).rawValue,
                                 Key.resultsCount: "\(resultsCount)",
                                 Key.millisecondsSinceRequestSent: "\(millisecondsSinceRequestSent)"
                               ])
         }
     }
 }
 
-private extension WooAnalyticsEvent.PointOfSale.CartItemType {
-    var analyticsValue: String {
-        switch self {
-        case .simpleProduct:
-            return "simple"
-        case .variation:
-            return "variation"
+extension WooAnalyticsEvent.PointOfSale {
+    /// Source of the event where the event is triggered
+    /// Views: Product, Variation, and Coupon Lists. Cart view and Checkout error.
+    ///
+    enum Source: String {
+        case product
+        case variation
+        case coupon
+        case cart
+        case error
```

Comment:
nit: the error case could be more specific as many views have an error state, if it means the "checkout error" in the comment.
```suggestion
        case checkoutError
```